lib/spack/spack/report.py

```diff
@@ -44,8 +44,8 @@ def fetch_package_log(pkg):
 
 
 class InfoCollector(object):
-    """Decorates PackageInstaller._install_task, which is called by
-    PackageBase.do_install for each spec, to collect information
+    """Decorates PackageInstaller._install_task, which is called via
+    PackageBase.do_install for individual specs, to collect information
```

Comment:
Does this mean we've lost the ability to do cdash/junit reports on builds of entire environments?

Reply:
Are reports for environments supported currently? In our deployment, [I had to shuffle things around a little in the install command](https://github.com/BlueBrain/spack/commit/411c847189a68266954c90a23e7d9e5a32faceed#diff-5c111ba1f5bfb321d12a190c2a79b33c7b53e3d02f6d0201f1703432b71cb17c) to get JUnit reports to show up, and then take care to not have Spack processes overwrite each others reports. Also seems like the resulting reports will need some de-duplication.

Reply:
> Does this mean we've lost the ability to do cdash/junit reports on builds of entire environments?

AFAICT The only thing I ended up changing to this file in this PR is the comment.  

We *used* to collect data based on the package `do_install` method because the build would call each dependency package's `do_install` (so their data would be collected).  

The distributed build work involved a refactor to use a build queue.  This basically resulted in individual packages getting installed by the `_install_task`, not `do_install` method, in order to continue to collect the data on dependency packages.  Hence the switch described in the comment.

I haven't tested report collection on environments so I'll try that.

Reply:
> Are reports for environments supported currently? In our deployment, [I had to shuffle things around a little in the install command](https://github.com/BlueBrain/spack/commit/411c847189a68266954c90a23e7d9e5a32faceed#diff-5c111ba1f5bfb321d12a190c2a79b33c7b53e3d02f6d0201f1703432b71cb17c) to get JUnit reports to show up, and then take care to not have Spack processes overwrite each others reports. Also seems like the resulting reports will need some de-duplication.

Interesting.  Looks like we might need a bit of your shuffle added back into the build.

Reply:
Confirmed that the report does not appear to be generated.

Reply:
> > Are reports for environments supported currently? In our deployment, [I had to shuffle things around a little in the install command](https://github.com/BlueBrain/spack/commit/411c847189a68266954c90a23e7d9e5a32faceed#diff-5c111ba1f5bfb321d12a190c2a79b33c7b53e3d02f6d0201f1703432b71cb17c) to get JUnit reports to show up, and then take care to not have Spack processes overwrite each others reports. Also seems like the resulting reports will need some de-duplication.
> 
> Interesting. Looks like we might need a bit of your shuffle added to the main line.

@matz-e I added your "shuffling" in my latest commit (0d7ce1f) and confirmed at least the `junit` output is now generated for an environment.

Reply:
Thanks, glad it's of some use!